lib/block_view_message.go

```diff
@@ -825,95 +949,189 @@ func (bav *UtxoView) _connectMessagingGroup(
+				return 0, 0, nil, errors.Wrapf(err, "_connectMessagingGroup: "+
+					"Problem validating public key or messaging key for messagingMember (%v)", messagingMember.GroupMemberPublicKey[:])
+			}
 
-	// Validate all members.
-	for _, messagingMember := range txMeta.MessagingGroupMembers {
-		// Encrypted public key cannot be empty, and has to have at least as many bytes as a generic private key.
-		//
-		// Note that if someone is adding themselves to an unencrypted group, then this value can be set to
-		// zeros or G, the elliptic curve group element, which is also OK.
-		if len(messagingMember.EncryptedKey) < btcec.PrivKeyBytesLen {
-			return 0, 0, nil, errors.Wrapf(
-				RuleErrorMessagingMemberEncryptedKeyTooShort, "_connectMessagingGroup: "+
-					"Problem validating messagingMember encrypted key for messagingMember (%v): Encrypted "+
-					"key length %v less than the minimum allowed %v. If this is an unencrypted group "+
-					"member, please set %v zeros for this value", messagingMember.GroupMemberPublicKey[:],
-				len(messagingMember.EncryptedKey), btcec.PrivKeyBytesLen, btcec.PrivKeyBytesLen)
+			// Now make sure messagingMember's MessagingGroupKey has already been added to UtxoView or DB.
+			// We encrypt the groupMessagingKey to recipients' messaging keys.
+			memberMessagingGroupKey := NewMessagingGroupKey(
+				messagingMember.GroupMemberPublicKey, messagingMember.GroupMemberKeyName[:])
+			memberGroupEntry := bav.GetMessagingGroupKeyToMessagingGroupEntryMapping(memberMessagingGroupKey)
+			// The messaging key has to exist and cannot be deleted.
+			if memberGroupEntry == nil || memberGroupEntry.isDeleted {
+				return 0, 0, nil, errors.Wrapf(
+					RuleErrorMessagingMemberKeyDoesntExist, "_connectMessagingGroup: "+
+						"Problem verifying messaing key for messagingMember (%v)", messagingMember.GroupMemberPublicKey[:])
+			}
+			// The messagingMember can't be already added to the list of existing members.
+			if _, exists := existingMembers[*messagingMember.GroupMemberPublicKey]; exists {
+				return 0, 0, nil, errors.Wrapf(
+					RuleErrorMessagingMemberAlreadyExists, "_connectMessagingGroup: "+
+						"Error, messagingMember already exists (%v)", messagingMember.GroupMemberPublicKey[:])
+			}
+			// Add the messagingMember to our helper structs.
+			existingMembers[*messagingMember.GroupMemberPublicKey] = true
+			newMessagingMembers = append(newMessagingMembers, messagingMember)
 		}
 
-		// Make sure the messagingMember public key and messaging key name are valid.
-		if err := ValidateGroupPublicKeyAndName(messagingMember.GroupMemberPublicKey[:], messagingMember.GroupMemberKeyName[:]); err != nil {
-			return 0, 0, nil, errors.Wrapf(err, "_connectMessagingGroup: "+
-				"Problem validating public key or messaging key for messagingMember (%v)", messagingMember.GroupMemberPublicKey[:])
+	case MessagingGroupOperationMuteMembers:
+		// Muting members assumes the group was already created.
```

Comment:
I think we need to check block height here before proceeding

Reply:
The call to GetMessagingGroupOperation() is gated by the blockheight, so the operation will never be Mute/Unmute unless we're past the block height.